demo/client.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# This is generated code!
+
+module Weather
+  class Client < Smithy::Client::Base
+    self.service_shape = Shapes::WEATHER
+
+    add_plugin(Smithy::Client::Plugins::NetHTTP)
+
+
+    # TODO
+    def initialize(*args)
+      super(*args)
+    end
+
+    # TODO!
+    def get_city(params = {}, options = {})
+      input = build_input(:get_city, "example.weather#GetCity", params)
+      input.send_request(options)
+    end
+
+    # TODO!
+    def get_current_time(params = {}, options = {})
+      input = build_input(:get_current_time, "example.weather#GetCurrentTime", params)
+      input.send_request(options)
+    end
+
+    # TODO!
+    def get_forecast(params = {}, options = {})
+      input = build_input(:get_forecast, "example.weather#GetForecast", params)
+      input.send_request(options)
+    end
+
+    # TODO!
+    def list_cities(params = {}, options = {})
+      input = build_input(:list_cities, "example.weather#ListCities", params)
+      input.send_request(options)
+    end
+
+    private
+
+    def build_input(operation_name, operation_id, params)
+      handlers = @handlers.for(operation_name)
+      context = Smithy::Client::HandlerContext.new(
+        operation_name: operation_name,
+        operation: config.service_shape.operation(operation_id),
```

Comment:
Yeah, this is a good general question.  I could see us using: `operation` for the name and `operation_shape` for the shape.  OR `operation` for the shape and `operation_name` for the name...

But - we don't really need both name and shape right?  We really only need the operation shape since that can tell us the name.  So I actually lean towards just having a single member `operation` (that is the shape)

Reply:
I think I'm just going to leave this as it is for now.